enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/readreplica/NoOpLockManager.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) 2002-2017 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.causalclustering.readreplica;
+
+import java.util.stream.Stream;
+
+import org.neo4j.kernel.impl.locking.ActiveLock;
+import org.neo4j.kernel.impl.locking.LockTracer;
+import org.neo4j.kernel.impl.locking.Locks;
+import org.neo4j.storageengine.api.lock.AcquireLockTimeoutException;
+import org.neo4j.storageengine.api.lock.ResourceType;
+
+public class NoOpLockManager implements Locks
+{
+    @Override
+    public Locks.Client newClient()
+    {
+        return new Client();
+    }
+
+    @Override
+    public void accept( Visitor visitor )
+    {
+    }
+
+    @Override
+    public void close()
+    {
+    }
+
+    private class Client implements Locks.Client
+    {
+        @Override
+        public void acquireShared( LockTracer tracer, ResourceType resourceType, long... resourceIds ) throws AcquireLockTimeoutException
+        {
+        }
+
+        @Override
+        public void acquireExclusive( LockTracer tracer, ResourceType resourceType, long... resourceIds ) throws AcquireLockTimeoutException
+        {
+            throw new IllegalStateException();
```

Comment:
Yea it is interesting what to do here really, because replication happens behind the scenes regardless. None of these locks will have a distributed effect. Neither an exclusive nor a shared lock will be able to prevent upstream updates coming in. I am not really sure which semantics to aim for here. Would be better if all instances participated in concurrency control properly.

This is trying to catch the case early of someone doing a writing query on a read replica.